geniza/entities/models.py

```diff
@@ -68,6 +70,78 @@ def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
 
 
+class Event(models.Model):
+    """A historical event involving one or more documents, and optionally additional entities."""
+
+    name = models.CharField(
+        max_length=255,
+        help_text="A short name for the event to use as a display label",
+    )
+    description = models.TextField(
+        blank=True,
+        help_text="A description of the event",
+    )
+    footnotes = GenericRelation(Footnote, blank=True, related_query_name="event")
+    display_date = models.CharField(
+        "Display date",
+        help_text='''Display date for manually entered or automatically generated date,
+        as it should appear in the public site, such as "Late 12th century"''',
+        max_length=255,
+        blank=True,  # use standard date for display if this is blank
+    )
+    standard_date = models.CharField(
+        "CE date override",
+        help_text="Manual override for automatically generated date or date range. "
+        + DocumentDateMixin.standard_date_helptext,
+        blank=True,  # use automatic date range from associated documents if this is blank
+        null=True,
+        max_length=255,
+        validators=[RegexValidator(DocumentDateMixin.re_date_format)],
+    )
+
+    def __str__(self):
+        """Use the name field for string representations"""
+        return self.name
+
+    @property
+    def date_str(self):
+        """Return a formatted string for the event's date/range, for use in public site"""
+        return (
+            self.display_date
+            or Document.standard_date_display(self.standard_date)
+            or Document.standard_date_display(self.documents_date_range)
+        )
+
+    @property
+    def documents_date_range(self):
+        """Standardized range of dates across associated documents"""
+        full_range = [None, None]
+        # NOTE: This looks similar to Document.dating_range() but applies across multiple docs
+        for doc in self.documents.all():
+            doc_range = doc.dating_range()
+            if doc_range and doc_range[0]:
+                start = doc_range[0]
+                end = doc_range[1] if len(doc_range) > 1 else start
+                # use numeric format to compare to current min, replace if smaller
+                start_numeric = int(start.numeric_format(mode="min"))
+                min = full_range[0]
+                if min is None or start_numeric < int(min.numeric_format(mode="min")):
+                    # store as PartialDate
+                    full_range[0] = start
+                # use numeric format to compare to current max, replace if larger
+                end_numeric = int(end.numeric_format(mode="max"))
+                max = full_range[1]
+                if max is None or end_numeric > int(max.numeric_format(mode="max")):
+                    # store as PartialDate
+                    full_range[1] = end
```

Comment:
I had to look back at the other one to remind myself; I remember some of our discussion on that before (and wish I had a better solution to offer you).  

the one thing I wonder is if this could be a utility method that you call in both places and call it on the same content - instead of calling here on the result of dating range, could you call in both places on the dating set instances? (either for one document or all of them).

My other question is a conceptual / semantic one: does any date always take precedence over `None` for either start/end bounds?

Reply:
> the one thing I wonder is if this could be a utility method that you call in both places and call it on the same content - instead of calling here on the result of dating range, could you call in both places on the dating set instances? (either for one document or all of them).

Not exactly that, since we need to also consider each of the documents' `document_date` and not just their inferred `Dating` sets.  However, the comparison between two `PartialDate` ranges ends up being essentially the same code, so that can be factored out into a utility method! (I moved it to a static method on `PartialDate` so that it's clear the arguments should both be sets of `PartialDate` instances.)

> My other question is a conceptual / semantic one: does any date always take precedence over None for either start/end bounds?

Not sure I fully understand, but I think the answer is yes; we don't support unbounded date ranges, either on-document or inferred.